### ISSUE TITLE
Not attempting to connect from inside docker container

### ISSUE THREAD

**Issue by FriwiDev** [couchbase/couchbase-lite-java-ce-root#9]
Version: 2.7.1

We are currently stuck on a project that should run a couchbase lite instance in order to notify another service of updates. We use docker for our complete setup. When running the project from our IDE on Windows 10, we experience no issues. Even when the sync gateway is down/unreachable the library happily tries to reconnect to it. On docker with openjdk:8-jdk-alpine base image, we fail to even get a connection attempt. The log simply ends at...

`I/CouchbaseLite/REPLICATOR:Replicator{@51016012,<*,Database{@3d04a311, name='output-dd'},URLEndpoint{url=ws://127.0.0.1:4984/output-dd}] is connecting, progress 0/0, error: null`

...and stays there forever. I've included maven build info, Dockerfile, output logs in all scenarios and relevant source code in the following gist:

https://gist.github.com/FriwiDev/15e88553a0494df5ff860020bfb1b088

Any help or hints would be greatly appreciated! :)

**Comment by bmeike** [comment]
Wow... Sorry for talking so long to get to this!  Did you figure out what was going wrong, or find a workaround?

**Comment by FriwiDev** [comment]
The problem was that I was using an alpine linux docker image. I had to change to debian and include libicuuc in version 54 on the library path. Now everything is working flawless.

I guess the endless connecting state was caused because I installed gcompat to fix library errors in the alpine container. So I guess gcompat was not a good option to take.

As my problem is fixed, I will close this issue. Thank you for replying, I almost forgot to post the fix to this issue :)
